### ISSUE TITLE
Batoto

### ISSUE THREAD

**Issue by chrysalist** [fmd-project-team/FMD#955]
Doesn't fetch new chapters in favourites and can't access titles

**Comment by Slasar41** [comment]
Same issue as https://github.com/fmd-project-team/FMD/issues/954.
Try follow the instruction on the comment.

**Comment by chrysalist** [comment]
I followed the instructions and did the steps a couple of times but it's still not working. Will try again after a while and post the progress here

**Comment by Slasar41** [comment]
Try copying all the cookies string to FMD.
![image](https://user-images.githubusercontent.com/28888646/80580956-91f76c80-8a36-11ea-808b-f9db467dfe4c.png)
![image](https://user-images.githubusercontent.com/28888646/80581021-b6ebdf80-8a36-11ea-8c89-c135eddb9eea.png)

**Comment by chrysalist** [comment]
Thank u for helping me😊 I used chrome this time and it seems to have worked. I can access titles and download the chapters. But now I have encountered another problem. When I check the titles for new chapters, it keeps showing me that I have new chapters to download. Pls help me for one last time🙇🏻‍♀️

**Comment by chrysalist** [comment]
![Screenshot (3)](https://user-images.githubusercontent.com/47241085/80594333-7002e780-8a40-11ea-8351-8bcf0c377e83.png)

**Comment by Slasar41** [comment]
Huh, first time seeing it.
Maybe there's really a new chapters?
Try just enable some of it and let it finish checking if it'll download something or not.

**Comment by chrysalist** [comment]
Nah. I download manga everyday so everything is always updated. And I ran the check twice. Everytime it shows the same thing. I downloaded some of the chapters from some of the titles. They don't show up when I check for new chapters thankfully. It seems that the download bookmark is somehow removed the

**Comment by HiP1** [comment]
I did the cloudlare workaround several times, and the chapter list never shows up. so i can't download from batoto anymore.
though i can update the manga list.

**Comment by HiP1** [comment]
tried several browsers, and changing the cookies order and format, still nothing. not even the manga info or favorite check. in the log, there is the ReferenceError to 'a' as undefined. same as Mangago

**Comment by Slasar41** [comment]
@HiP1 Try delete the website module and let FMD redownload it.

**Comment by HiP1** [comment]
> 
> 
> @HiP1 Try delete the website module and let FMD redownload it.

thanks for the suggestion, but it didn't work.
when i try to update the manga list, i have to specify the directory page number, but even then, i get the "Error: ReferenceError: identifier 'a' undefined" in the log for every page too.
even if i comment out any reference to "a" in the lua, the error still appears, so i think there is something wrong somewhere else, maybe in the pascal ?

**Comment by HiP1** [comment]
does Batoto work for anyone ? can't seem to make it work at all.

**Comment by chrysalist** [comment]
It does for me. But there's one annoying problem. I have to add the cookies and user agent codes form batoto again and again when it stops working. I had to do this 2-3 times these last few days.

**Comment by HiP1** [comment]
> 
> 
> It does for me. But there's one annoying problem. I have to add the cookies and user agent codes from batoto again and again when it stops working. I had to do this 2-3 times these last few days.

thanks for your reply. could you detail your process to make it work ?
i tried chrome in incognito, firefox, edge user agents, and copying all or only cloudflare cookies or a random selection of cookie. nothing worked

**Comment by chrysalist** [comment]
I will tell u the steps that I do. 
1. Go to bato.to
2. Let it load. After that I click f12 and then f5. 
3. I click on the topmost one i.e usually bato.to. I get the user agent first and then all the cookies that r there. 
4. Aftrr

**Comment by HiP1** [comment]
> 
> 
> I will tell u the steps that I do.
> 
>     1. Go to bato.to
> 
>     2. Let it load. After that I click f12 and then f5.
> 
>     3. I click on the topmost one i.e usually bato.to. I get the user agent first and then all the cookies that r there.
> 
>     4. After that I click on thttp setting right on top of the cookies box ( I don't know if its an actual relevant action but it seems to work when I do that.)
> 
>     5. Then click apply
>        It seems to work after that.
>        I don't make any kind of changes in the cookies. I copy and just paste them all

Thanks for taking the time to explain. I did all that step by step, but it didn't work, unfortunately.
which browser and version are you using ?

**Comment by chrysalist** [comment]
I use the latest version of chrome. I tried to do it with firefox before but it didn't work.

**Comment by HiP1** [comment]
> 
> 
> I use the latest version of chrome. I tried to do it with firefox before but it didn't work.

i'm on the dev channel and canary version of chrome. maybe that's why. but it works in the browser, it should work in FMD too. weird

**Comment by chrysalist** [comment]
Hmmm🤔. It should work then. Maybe try doing it with the latest version of chrome once. We can compare them to see if there is any difference.

**Comment by dazedcat19** [comment]
try fmd2. I just pushed a cloudflare fix.

**Comment by HiP1** [comment]
> 
> 
> try fmd2. I just pushed a cloudflare fix.
oh interesting, i'll try
I see your own project FMD2 and another FMD2 that was from august 2019 on github, both forked from here. are those related at all ?

**Comment by HiP1** [comment]
> 
> 
> try fmd2. I just pushed a cloudflare fix.

oh interesting.
I see two FMD2 projects here on github, both forked from the one here. one in your own projects, and another from "FreeMangaDownloader", are they related at all ?
should I just use yours ?

**Comment by HiP1** [comment]
> 
> 
> Hmmm🤔. It should work then. Maybe u should try doing it with the latest version of chrome once. We can compare them to see if there is any difference.

might take some time, as i need the chrome dev channel and canary. and it should work with those as they use their own user agent and it works in browser, just not in FMD. I dont see why it would work in browser and not in FMD if they use the same user agent string.

**Comment by HiP1** [comment]
> 
> 
> try fmd2. I just pushed a cloudflare fix.

your own FMD2 project worked for Batoto.
it did check favorites and download chapters.

**Comment by FirstDagger** [comment]
> try fmd2. I just pushed a cloudflare fix.

Can confirm your [Cloudflare fix](https://github.com/dazedcat19/FMD2) works for me and allows Batoto to be searched and downloaded again.
